app/questionnaire/questionnaire_schema.py

```diff
@@ -211,6 +211,16 @@ def get_section_ids_dependent_on_list(self, list_name: str) -> list[str]:
             self._list_name_to_section_map[list_name] = section_ids
             return section_ids
 
+    def get_section_ids_dependent_on_when_rules(
+        self, section: ImmutableDict
+    ) -> set[str]:
+        when_rules = self._get_values_for_key(section, "when")
+        rules: Union[dict, list] = next(when_rules, [])
+        section_ids_dependent_on_rules = self._get_section_ids_dependent_on_rules(
+            section["id"], rules
+        )
+        return section_ids_dependent_on_rules
```

Comment:
```suggestion
        return self._get_section_ids_dependent_on_rules(
            section["id"], rules
        )
```

Reply:
Always a bit or personal choice, but I am not fussy so happy to change